packages/cmr-client/CMRSearchConceptQueue.js

```diff
@@ -0,0 +1,81 @@
+'use strict';
+
+const searchConcept = require('./searchConcept');
+
+/**
+ * A class to efficiently list all of the concepts (collections/granules) from
+ * CMR search, without loading them all into memory at once.  Handles paging.
+ *
+ * @typicalname cmrSearchConceptQueue
+ *
+ * @example
+ * const { CMRSearchConceptQueue } = require('@cumulus/cmr-client');
+ *
+ * const cmrSearchConceptQueue = new CMRSearchConceptQueue(
+ *   'my-provider',
+ *   'my-clientId',
+ *   'granule',
+ *   {},
+ *   'json'
+ * );
+ */
+class CMRSearchConceptQueue {
+  /**
+   * The constructor for the CMRSearchConceptQueue class
+   *
+   * @param {Object} params
+   * @param {string} params.provider - the CMR provider id
+   * @param {string} params.clientId - the CMR clientId
+   * @param {string} params.type - the type of search 'granule' or 'collection'
+   * @param {string} [params.searchParams={}] - the search parameters
+   * @param {string} params.format - the result format
+   */
+  constructor(params = { searchParams: {} }) {
```

Comment:
I think it's not necessary to have default value here `= { searchParams: {} }`
Should we check if the mandatory params keys are present?

Reply:
Since CMRSearchConceptQueue now takes a parameter object instead of a list of parameters, the code uses `new CMRSearchConceptQueue` should be updated (e.g. create-reconciliation-report.js)

Reply:
This should actually be okay because create-reconciliation-report.js still calls the CMRSearchConceptQueue in @cumulus/cmrjs which takes care of the change in param structure before it calls this one. 
here: https://github.com/nasa/cumulus/pull/842/files#diff-edfb462e041ef85a0cacf11443f5e20dR193

Reply:
Why do we keep the `CMRSearchConceptQueue ` in both places?  CMRSearchConceptQueue is used by cumulus internally.

Reply:
You make a good point there. Maybe we could keep CMRSearchConceptQueue in cmrjs. @yjpa7145  thoughts?

Reply:
The goal with creating the `cmr-client` package was to put all of the code that calls CMR in one place, but without the additional dependencies that `cmrjs` contained (aws-sdk and @cumulus/common, mainly).  By doing that, other projects like cumulus-dashboard could use the `cmr-client` package.  CMRSearchConceptQueue is certainly something that another project like cumulus-dashboard may want to use.

As for having it in two places, that's to maintain backward compatibility for the time being.  The expectation is that we'll update any of our code that uses CmrSearchConceptQueue to use the one provided by `cmr-client`, add deprecation warnings to the one in `cmrjs`, and then remove it from `cmrjs` in the future.

Reply:
Added the deprecated in cmrjs

Reply:
Correct me if I'm wrong, but I thought the architecture meeting where we discussed release woes it was decided that deprecating should involve updating all the references? `CMRSearchConceptQueue` from `cmrjs` is still referenced in a lot of places, for example.

Reply:
That doesn't have to happen in this PR, but it shouldn't fall through the cracks.